src/pkg/scan/postprocessors/report_converters.go

```diff
@@ -0,0 +1,207 @@
+	}
+
+	// parse the raw report with the V1 schema of the report to the normalized structures
+	rawReport := new(vuln.Report)
+	if err := json.Unmarshal([]byte(reportData), &rawReport); err != nil {
+		return "", "", errors.Wrap(err, fmt.Sprintf("Error when toSchema V1 report to V2"))
+	}
+
+	if err := c.toSchema(ctx, reportUUID, registrationUUID, digest, reportData); err != nil {
+		return "", "", errors.Wrap(err, fmt.Sprintf("Error when converting vulnerability report"))
+	}
+	rawReport.Vulnerabilities = nil
+	data, err := json.Marshal(rawReport)
+	if err != nil {
+		return "", "", errors.Wrap(err, fmt.Sprintf("Error when persisting raw report summary"))
+	}
+	return reportUUID, string(data), nil
+
+}
+
+// FromRelationalSchema converts the generic vulnerability record stored in relational form to the
+// native JSON blob.
+func (c *nativeToRelationalSchemaConverter) FromRelationalSchema(ctx context.Context, reportUUID string, artifactDigest string, reportSummary string) (string, error) {
+	vulns, err := c.dao.GetForReport(ctx, reportUUID)
+	if err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("Error when toSchema generic vulnerability records for %s", reportUUID))
+	}
+	rp, err := c.fromSchema(ctx, reportUUID, artifactDigest, reportSummary, vulns)
+	if err != nil {
+		return "", err
+	}
+	return rp, nil
+}
+
+func (c *nativeToRelationalSchemaConverter) toSchema(ctx context.Context, reportUUID string, registrationUUID string, digest string, rawReportData string) error {
+
+	var vulnReport vuln.Report
+	err := json.Unmarshal([]byte(rawReportData), &vulnReport)
+	if err != nil {
+		return err
+	}
+	for _, v := range vulnReport.Vulnerabilities {
+		vulnV2 := new(scan.VulnerabilityRecord)
+		vulnV2.CVEID = v.ID
+		vulnV2.Description = v.Description
+		vulnV2.Package = v.Package
+		vulnV2.PackageVersion = v.Version
+		vulnV2.PackageType = "Unknown"
+		vulnV2.Fix = v.FixVersion
+		vulnV2.URLs = strings.Join(v.Links, "|")
+		vulnV2.RegistrationUUID = registrationUUID
+		vulnV2.Severity = v.Severity.String()
+
+		// process the CVSS scores if the data is available
+		if (vuln.CVSS{} != v.CVSSDetails) {
+			vulnV2.CVE3Score = v.CVSSDetails.ScoreV3
+			vulnV2.CVE2Score = v.CVSSDetails.ScoreV2
+			vulnV2.CVSS3Vector = v.CVSSDetails.VectorV3
+			vulnV2.CVSS2Vector = v.CVSSDetails.VectorV2
+		}
+		if len(v.CWEIds) > 0 {
+			vulnV2.CWEIDs = strings.Join(v.CWEIds, ",")
+		}
+
+		// marshall the presented vendor attributes as a json string
+		if len(v.VendorAttributes) > 0 {
+			vendorAttributes, err := json.Marshal(v.VendorAttributes)
+			// set the vendor attributes iff unmarshalling is successful
+			if err == nil {
+				vulnV2.VendorAttributes = string(vendorAttributes)
+			}
+		}
+		o, err := orm.FromContext(ctx)
+		if err != nil {
+			return err
+		}
+		err = o.Begin()
+		if err != nil {
+			return err
+		}
+		_, err = c.dao.InsertForReport(ctx, reportUUID, vulnV2)
+		if err != nil {
+			o.Rollback()
+			return err
+		}
+		o.Commit()
```

Comment:
Don't begin a transaction manually because the ormer from the context already began in the http request handler.

Please use this way to handle the transaction when you need.

```go

insertForReport := func(ctx context.Context) error {
     _, err = c.dao.InsertForReport(ctx, reportUUID, vulnV2)
    return err
}

err := orm.WithTransaction(insertForReport)(ctx)

if err != nil && errors.IsConflictErr(err) {
    return err 
}
```

Reply:
Done. Added transaction support